mypy/fastparse.py

```diff
@@ -67,6 +67,13 @@
 TYPE_COMMENT_AST_ERROR = 'invalid type comment or annotation'
 
 
+# Older versions of typing don't allow using overload outside stubs,
+# so provide a dummy.
+if sys.version_info < (3, 6):
```

Comment:
Shouldn't this be guarded with a big `if not MYPY:`?

Reply:
mypy didn't seem to care, but probably ought to anyways

Reply:
On 3.4, the version of typing is newer than on 3.5, so you lose some type safety on 3.4. It works because you are only making the signature less exact, but it is more correct to use `if MYPY`.